XForm/XForm.Test/Functions/FunctionsTests.cs

```diff
@@ -20,6 +17,63 @@ public class FunctionsTests
     {
         public static DateTime TestAsOfDateTime = new DateTime(2017, 12, 10, 0, 0, 0, DateTimeKind.Utc);
 
+        [TestMethod]
+        public void Function_Coalesce()
+        {
+            int[] ageValues = { 50, 23, 42, 0, 0, 0 };
+            int[] salaryValues = { 5, 0, 12, 0, 13, 0 };
+            int[] siblingValues = { 0, 0, 0, 1, 0, 2 };
```

Comment:
Maybe throw in a few non-default value nulls (-1 for value). I know you don't, but just checking that code doesn't assume nulls will always have the value default(T).